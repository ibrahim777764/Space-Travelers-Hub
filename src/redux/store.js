/* eslint-disable import/order */

import {
  applyMiddleware,
  combineReducers,
  compose,
  createStore,
} from 'redux';

import logger from 'redux-logger';
import missionsReducer from './missions/missions';
import rocketsReducer from './rockets/rockets';
import thunk from 'redux-thunk';

let composeEnhancers = null;
if (process.env.NODE_ENV === 'development') {
  composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
} else {
  composeEnhancers = compose;
}
const reducer = combineReducers({
  missionsReducer, rocketsReducer,
});

const store = createStore(
  reducer,
  composeEnhancers(applyMiddleware(logger, thunk)),
);

export default store;
