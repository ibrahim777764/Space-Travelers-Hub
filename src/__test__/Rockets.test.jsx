import '@testing-library/jest-dom';

import { Provider } from 'react-redux';
import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import RocketList from '../components/Rockets';
import store from '../redux/store';

jest.mock('../components/Rockets');

describe('Group of tests for Rockets page', () => {
  it('renders rockets list correctly', () => {
    const list = renderer
      .create(<Provider store={store}><RocketList /></Provider>)
      .toJSON();
    expect(list).toMatchSnapshot();
  });

  test('List is created', () => {
    const { container } = render(<Provider store={store}><RocketList /></Provider>);
    const list = container.querySelector('.container-fluid');
    expect(list).toBeInTheDocument();
    expect(container).toHaveTextContent('The Falcon 1');
    expect(container).toHaveTextContent('by SpaceX during 2006-2009');
  });

  test('The Rocket discription should have the words Falcon, SpaceX, etc...', () => {
    const { container } = render(<Provider store={store}><RocketList /></Provider>);
    expect(container).toHaveTextContent('The Falcon 1');
    expect(container).toHaveTextContent('by SpaceX during 2006-2009');
  });

  test('If reserved = "true" button text should be, Cancel reservation', () => {
    const { container } = render(<Provider store={store}><RocketList /></Provider>);
    const listbtn2 = container.querySelector('.btn2');
    expect(listbtn2).toHaveTextContent('Cancel reservation');
  });
});
