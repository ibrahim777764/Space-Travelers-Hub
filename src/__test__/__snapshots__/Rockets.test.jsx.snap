// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group of tests for Rockets page renders rockets list correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="container-fluid d-flex my-4 mx-0 mainDiv"
  >
    <img
      alt="Falcon 1"
      className="img-fluid"
      src="https://imgur.com/DaCfMsj.jpg"
    />
    <div>
      <h3>
        Falcon 1
      </h3>
      <div>
        <button
          bg="info"
          type="button"
        >
          Reserved
        </button>
        <p>
          The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
        </p>
      </div>
      <button
        className="btn2"
        type="button"
        variant="outline-danger"
      >
        Cancel reservation
      </button>
    </div>
  </div>
  <div
    className="container-fluid d-flex my-4 mx-0 mainDiv"
  >
    <img
      alt="Falcon 1"
      className="img-fluid"
      src="https://imgur.com/DaCfMsj.jpg"
    />
    <div>
      <h3>
        Falcon 1
      </h3>
      <div>
        <button
          bg="info"
          type="button"
        >
          
        </button>
        <p>
          The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
        </p>
      </div>
      <button
        className="btn2"
        type="button"
        variant="outline-secondary"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
  <div
    className="container-fluid d-flex my-4 mx-0 mainDiv"
  >
    <img
      alt="Falcon 1"
      className="img-fluid"
      src="https://imgur.com/DaCfMsj.jpg"
    />
    <div>
      <h3>
        Falcon 1
      </h3>
      <div>
        <button
          bg="info"
          type="button"
        >
          Reserved
        </button>
        <p>
          The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
        </p>
      </div>
      <button
        className="btn2"
        type="button"
        variant="outline-danger"
      >
        Cancel reservation
      </button>
    </div>
  </div>
  ;
</div>
`;
